crates/generated_parser/src/ast_builder.rs

```diff
@@ -4648,6 +4808,9 @@ impl<'alloc> AstBuilder<'alloc> {
                         &self.atoms.borrow(),
                     )?;
                 }
+                // Do nothing for Labels, as they have to be nested for
+                // a syntax error to occur
```

Comment:
what does "they have to be nested for a syntax error to occur" mean?

Reply:
I should make it clearer. The error occurs in this case:

```
L1: L1: "foo"
```
But not in this case:
```
L1: "foo"
L1: "bar"
```

We do not have the nesting information at the script/block/function level so we cannot reuse the mechanism used for checking duplicate bindings in those error contexts. We only know that the label occurred somewhere in the block, and that it might occur more than once, not how it occurs. So, this is handled in `check_labelled_statement`.

I will add that as a comment.